arangod/GeoIndex/Index.cpp

```diff
@@ -187,7 +197,7 @@ S2LatLng Index::parseGeoDistance(aql::AstNode const* args,
     if (json.isArray() && json.length() >= 2) {
       res = shape.parseCoordinates(json, /*GeoJson*/ true);
     } else {
-      res = geo::geojson::parseRegion(json, shape);
+      res = geo::geojson::parseRegion(json, shape, false);
```

Comment:
so the legacy boolean here would be false because there would be no such a case for arrays, but does it have anything to do with _legacyPolygons being set to true or these values are not correlated?

Reply:
This is an interesting one. This code is only used when we set up an IndexIterator for a geo index. It is only used for the object `obj` in a statement like

```
FILTER GEO_DISTANCE(obj, d.geo) <= 10000
```

or so. Thus, it does not influence the way geo data in the database is parsed for indexing. That is why I intentionally put `false` here.

However, it does ever so slightly influence the centroid of `obj` which is used for the actual distance calculation when the query is executed. So therefore, I think you have a point, if we want to be fully backwards compatible for our queries, we would have to use `_legacyPolygons` here, and I will change this.
It is slightly painful, since this is a static member function, so we must hand the argument in, but this is doable.

Reply:
Done.